bpf/lib/overloadable_skb.h

```diff
@@ -43,8 +48,12 @@ get_epid(const struct __sk_buff *ctx)
 static __always_inline __maybe_unused void
 set_identity_mark(struct __sk_buff *ctx, __u32 identity)
 {
-	ctx->mark = ctx->mark & MARK_MAGIC_KEY_MASK;
-	ctx->mark |= ((identity & 0xFFFF) << 16) | ((identity & 0xFF0000) >> 16);
+	__u32 cluster_id = (identity >> IDENTITY_LEN) & CLUSTER_ID_MAX;
+	__u32 cluster_id_lower = cluster_id & 0xFF;
+	__u32 cluster_id_upper = ((cluster_id & 0xFFFFFF00) << (8 + IDENTITY_LEN));
+
+	ctx->mark &= MARK_MAGIC_KEY_MASK;
+	ctx->mark |= (identity & IDENTITY_MAX) << 16 | cluster_id_lower | cluster_id_upper;
```

Comment:
I think an ascii diagram representing the bits meaning would be useful.

Reply:
added :+1: